qiskit/pulse/transforms/channel_transforms.py

```diff
@@ -0,0 +1,330 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+from collections import defaultdict
+from typing import Optional, Union, Tuple, Iterable, Iterator, Dict, List
+
+import numpy as np
+
+from qiskit.pulse import channels
+from qiskit.providers.basebackend import BaseBackend
+from qiskit import pulse, circuit
+from qiskit.pulse.instructions import Instruction
+
+from base_transforms import target_qobj_transform
+from pulse_instruction_types import PhaseFreqTuple, PulseInstruction, OpaqueShape
+from device_info import OpenPulseBackendInfo
+
+InstructionSched = Union[Tuple[int, Instruction], Instruction]
+ScheduleLike = Union[pulse.Schedule, pulse.ScheduleBlock, 
+                     InstructionSched, Iterable[InstructionSched]]
+
+class ParsedInstruction:
+    def __init__(self, 
+                 t0: int,
+                 dt: float,
+                 frame: PhaseFreqTuple, 
+                 data: Union[Instruction, List[Instruction]], 
+                 is_opaque: bool, 
+                 xdata: np.ndarray = None,
+                 ydata: np.ndarray = None):
+        self.t0 = t0
+        self.dt = dt
+        self.frame = frame
+        self.data = data
+        self.is_opaque = is_opaque
+        self.xdata = xdata
+        self.ydata = ydata
+
+class ChannelTransforms:
+    """Channel transform manager."""
+
+    _waveform_group = (pulse.Play, pulse.Delay, pulse.Acquire)

```

Comment:
Maybe we can move this to below the `ScheduleLike` so that it can be easily found (when we add new instruction)?